src/main/java/com/xebialabs/xlrelease/ci/XLReleaseNotifier.java

```diff
@@ -222,54 +222,53 @@ public FormValidation doCheckXLReleaseClientProxyUrl(@QueryParameter String xlRe
-                }
-                return warning("Template does not exist.");
+                Release release = getTemplate(credential, template);
+                if (release != null)
+                    return FormValidation.okWithMarkup(Messages.XLReleaseNotifier_sucessmsg("Provided template path is valid."));
+                return error(Messages.XLReleaseNotifier_templateNotExist());
+            } catch (UniformInterfaceException exp) {
+                return error(Messages.XLReleaseNotifier_templateNotExist());
             } catch (Exception exp) {
-                return warning("Failed to communicate with XL Release server");
+                return error(exp.getMessage());
             }
         }
 
-        private Release getTemplate(String credential, String value) {
-            List<Release> candidates = getXLReleaseServer(credential).searchTemplates(value);
-            for (Release candidate : candidates) {
-                if (candidate.getTitle().equals(value)) {
-                    candidate.setVariableValues(getVariables(credential, candidate));
-                    return candidate;
-                }
-            }
-            return null;
-
-        }
-
         private Map<String, String> getVariables(String credential, Release release) {
+            if (release == null) {
+                return Collections.emptyMap();
+            }
             List<TemplateVariable> variables = getXLReleaseServer(credential).getVariables(release.getInternalId());
             return TemplateVariable.toMap(variables);
         }
 
-        public ListBoxModel doFillTemplateItems(@QueryParameter String credential) {
+        public AutoCompletionCandidates doAutoCompleteTemplate(@QueryParameter final String value) {
             try {
-                List<Release> templates = getXLReleaseServer(credential).getAllTemplates();
+                String queryString = value.replaceAll(SLASH_ESCAPE_SEQ, SLASH_MARKER);
+                String folderId = getFolderId(lastCredential, queryString);
 
-                @SuppressWarnings("unchecked")
-                Collection<String> titles = CollectionUtils.collect(templates, new Transformer() {
-                    public Object transform(Object o) {
-                        return ((Release) o).getTitle();
-                    }
-                });
+                List<Release> templates = getTemplatesByFolderID(lastCredential, folderId);
+                List<Folder> folders = getSubFolders(lastCredential, folderId);
+                final AutoCompletionCandidates autoCompletionCandidates = new AutoCompletionCandidates();
+
+                CollectionUtils.filter(folders, getFilterPredicate(getSearchString(queryString)));
+                CollectionUtils.filter(templates, getFilterPredicate(getSearchString(queryString)));
 
-                return of(titles);
+                for (Release template1 : templates)
```

Comment:
Can we please call variable 'template1' just 'template' or maybe 't', formatting logic is common for templates and folders move to common method